src/main/java/com/tecnalia/datausage/service/EnforcementService.java

```diff
@@ -360,4 +359,29 @@ void incrementAccessFrequency(String targetDataUri, String consumerUri) {
 		}
 		return;
 	}
+
+	/**
+	 * Get security profile from string.
+	 *
+	 * @param input The input value.
+	 * @return A security profile, if the value matches the provided enums.
+	 */
+	public static Optional<SecurityProfile> getSecurityProfile(final String input) {
```

Comment:
As mentioned earlier, with existing examples, here we have a null pointer exception.
If we put  ids:profile value in CA, value should be in this form:
 "ids:profile": {
        "@id": "https://w3id.org/idsa/code/BASE_SECURITY_PROFILE"
    }
Which was taken from infomodel 4.2.7. - de.fraunhofer.iais.eis.SecurityProfile.

And in that case, no matter what type of security profile is, the method will return Optional.Empty.